src/Schema/OracleSchemaManager.php

```diff
@@ -320,6 +320,109 @@ private function getQuotedIdentifierName($identifier)
         return $identifier;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        if ($currentDatabase === null) {
+            throw new Exception('Invalid database name.');
+        }
+
+        $tableNames = $this->listTableNames();
+
+        $objectsByTable = [];
```

Comment:
What's the purpose of this array? Could each collection of table-related columns/indices/etc. be stored in its own variable?

Reply:
Done